File: 2387-partition-array-such-that-maximum-difference-is-k/partition-array-such-that-maximum-difference-is-k.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)

        subsequences = 1
        min_val = nums[0]

        for i in range(1, n):
            if nums[i] - min_val > k:
                min_val = nums[i]
                subsequences += 1
        return subsequences

File: 2387-partition-array-such-that-maximum-difference-is-k/partition-array-such-that-maximum-difference-is-k.cpp
class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int n=nums.size();
        
        int subsequences = 1;
        int min_val = nums[0];
        for(int i=1;i<n;++i){
            if(nums[i]-min_val > k){
                min_val = nums[i];
                subsequences++;
            }
        }
        return subsequences;
    }
};